rmw_dds_common/test/test_graph_cache.cpp

```diff
@@ -1550,3 +1551,64 @@ TEST(test_graph_cache, bad_arguments)
     rcutils_reset_error();
   }
 }
+
+TEST(test_graph_cache, get_writers_info_by_topic_maybe_fail)
```

Comment:
@brawner would it make sense to split this test case into three separate ones (i.e. one for each method being fault injected).

Reply:
Yes, done! I think I originally intended for that given the test name.